autoload/lsp/omni.vim

```diff
@@ -451,3 +459,156 @@ function! s:get_refresh_pattern(server_names) abort
+
+            const expandable = get(completion_item, 'insertTextFormat', 1) == 2
+            if expandable
+                vim_complete_item['word'] = LspUtilsMakeValidWord(vim_complete_item['word'])
+                vim_complete_item['abbr'] = completion_item['label'] .. '~'
+            else
+                vim_complete_item['abbr'] = completion_item['label']
+            endif
+
+            if s:is_user_data_support
+                vim_complete_item['user_data'] = s:CreateUserData(completion_item, server_name, complete_position, start_characters[len(start_characters) - 1])
+            endif
+
+            vim_complete_items += [vim_complete_item]
+        endfor
+
+        # Add the additional text for startcol correction.
+        if start_character != default_start_character
+            for i in range(len(start_characters))
+                const item_start_character = start_characters[i]
+                if start_character < item_start_character
+                    const item = vim_complete_items[i]
+                    item['word'] = strcharpart(current_line, start_character, item_start_character - start_character) .. item['word']
+                endif
+            endfor
+        endif
+        var startcol = lsp#utils#position#lsp_character_to_vim('%', { 'line': complete_position['line'], 'character': start_character })
+
+        return { 'items': vim_complete_items, 'incomplete': incomplete ? 1 : 0, 'startcol': startcol }
+    enddef
+
+    def s:CreateUserData(completion_item: dict<any>, server_name: string, complete_position: dict<number>, start_character: number): string
+        const user_data_key = s:CreateUserDataKey(s:managed_user_data_key_base)
+        s:managed_user_data_map[user_data_key] = {
+        \   'complete_position': complete_position,
+        \   'server_name': server_name,
+        \   'completion_item': completion_item,
+        \   'start_character': start_character,
+        \ }
+        s:managed_user_data_key_base += 1
+        return user_data_key
+    enddef
+
+    def s:CreateUserDataKey(base: number): string
+        return '{"vim-lsp/key":"' .. base .. '"}'
+    enddef
+
+    " Replaced from lsp#utils#text_edit#get_range()
+    def s:LspUtilsTextEditGetRange(text_edit: any): any
+      if type(text_edit) != v:t_dict
+        return v:null
+      endif
+      const insert = get(text_edit, 'insert', v:null)
+      if type(insert) == v:t_dict
+        return insert
+      endif
+      return get(text_edit, 'range', v:null)
+    enddef
+
+    " Replaced from lsp#utils#make_valid_word().
+    def s:LspUtilsMakeValidWord(str: string): string
+       var s = substitute(str, '\$[0-9]\+\|\${\%(\\.\|[^}]\)\+}', '', 'g')
+       s = substitute(str, '\\\(.\)', '\1', 'g')
+       const valid = matchstr(str, '^[^"'' (<{\[\t\r\n]\+')
+       if empty(valid)
+           return s
+       endif
+       if valid =~# ':$'
+           return valid[ : -2]
+       endif
+       return valid
+    enddef
```

Comment:
These functions is replaced from external util functions. They are called frequency called.

As I mentioned in [this comment](https://github.com/prabirshrestha/vim-lsp/pull/1433/files#r1092647080), if use `vim9script` It will be able to use `import/export`.
To do so, needs to indent in legacy script.